schema.graphql

```diff
@@ -485,11 +544,78 @@ type MatchQuestionBuzz {
 """
 Docs group: Match
 """
-type MatchQuestionBonus {
+type BuzzPosition {
+  """
+  The time of the buzz. An ISO 8601-formatted string, with time zone, in the tournament's local time zone. May not be before the start time of reading the relevant question.
+  """
+  time: Date
+
+  """
+  The (`0`-based) index of the word on which the player buzzed.
   """
-  Which bonus number this was. May or may not match the `question_number` of the `MatchQuestion` parent, depending on whether bonuses are paired with tossups or simply read in order (and if the latter, on whether any bonus-earning questions failed to lead to a bonus).
+  word_index: Int
+
+  """
+  The (`0`-based) index of the clue on which the player buzzed. Clues may be defined in the `Question` object.
+  """
+  clue_index: Int
+}
+
+"""
+This object can be reused for every time the same question is read, i.e., it represents a question in general rather than a question being read in a specific match.
+
+Docs group: Match
+"""
+type Question {
+  """
+  The number of the question in the packet.
   """
   question_number: Int!
+  
+  """
+  The type of the question. We generally expect `tossup` or `bonus`, but are not restricting the values since some formats may use other types of questions.
+  """
+  type: String!
+  
+  """
+  The number of parts the question has (typically `1` if `type = "tossup"` and `3` if `type = "bonus"`, but some formats may vary).
+  """
+  parts: Int
+  
+  """
+  The identifier for the question in some external system, if any (e.g., the question IDs that NAQT prints below each question in its packet).
+  """
+  external_id: String
+  
+  """
+  Freeform information about the question's category, e.g. `British History`.
+  """
+  category: String
+  
+  """
+  Freeform information about the question's category at a higher level of aggregation, e.g. `History`.
+  """
+  category_group: String
```

Comment:
Does `category` and `subcategory` make more sense than `category_group` and `category`?

Reply:
I think people tend to talk more like "What category was that? Physics" than "What category was that? Science", but I don't feel strongly about it. I can also imagine adding more levels, or a true hierarchy, later.

Actually, should we just do that now? `category` could be an array of strings going up (or down) the hierarchy, e.g., `[20th-Century British History, British History, European History, History]`

Reply:
I think sticking with the two fields as they are here solves the problem we need to solve a little more easily.

Reply:
I'm fine with that. Do you feel strongly about `category` and `subcategory`?

Reply:
I don't.

Reply:
I usually use "category" for "Science" and subcategory for "Physics" (and subsubcategory for "Astronomy").

A freeform `category` field is okay, but I imagine we could benefit from some more standardization or structure, as categories are one of the most common targets of quizbowl data analysis I’ve seen recently. Perhaps a tree-like `Category` class? I use a simple one with just `name` and `parent` fields.